lib/trie/decode/byte.go

```diff
@@ -0,0 +1,16 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package decode
```

Comment:
does need to be separated out into their own packages? Seems like all the types in here are only used by the `trie` package so not really seeing the value in changing that.  Seperated into their own file makes sense, but I'm not sure we need another package for this.

Reply:
`Key` is used both in the leaf and branch packages, and `ReadNextByte` is used by `Key`, so I can't move it up in the `trie` package. I'm also aiming at reducing code in lib/trie since it's a rather large mix of things (all related to the trie, but still).

Reply:
Done in https://github.com/ChainSafe/gossamer/pull/2077/commits/30e2318a9ddfca712a319b3aecce07beaf252d85

`branch`, `leaf`, `encode` and `decode` are merged in `lib/trie/node` and it looks decent. Only key -> nibbles and nibbles -> key were moved to the new lib/trie/codec package since they're used in lib/trie as well and to avoid a confusing API.